pkg/execution/state/redis_state/key_generator.go

```diff
@@ -288,47 +288,54 @@ func (u queueKeyGenerator) ConcurrencyFnEWMA(fnID uuid.UUID) string {
 type BatchKeyGenerator interface {
 	// QueuePrefix returns the hash prefix used in the queue.
 	// This is likely going to be a redis specific requirement.
-	QueuePrefix() string
+	QueuePrefix(ctx context.Context, functionId uuid.UUID) string
 	// QueueItem returns the key for the hash containing all items within a
 	// queue for a function.  This is used to check leases on debounce jobs.
 	QueueItem() string
 	// BatchPointer returns the key used as the pointer reference to the
 	// actual batch
-	BatchPointer(context.Context, uuid.UUID) string
+	BatchPointer(ctx context.Context, functionId uuid.UUID) string
 	// BatchPointerWithKey returns the key used as the pointer reference to the
 	// actual batch for a given batchKey
-	BatchPointerWithKey(context.Context, uuid.UUID, string) string
+	BatchPointerWithKey(ctx context.Context, functionId uuid.UUID, key string) string
 	// Batch returns the key used to store the specific batch of
 	// events, that is used to trigger a function run
-	Batch(context.Context, ulid.ULID) string
+	Batch(ctx context.Context, isSharded bool, functionId uuid.UUID, batchId ulid.ULID) string
```

Comment:
Batch and BatchMetadata still require `isSharded` to be passed in to support older batches. This requirement drops after 60s of releasing this change, as all old batches will have been processed.